### ISSUE TITLE
🐛 [Bug Report] ores cannot be destroyed after prospecting

### ISSUE THREAD

**Issue by Nightingale9002** [cmangos/issues#3542]
### Bug Details

If you use Prospect on a item stack, and don't choose "auto loot all items" (at client config), and very quickly double click(or more, may need mouse marco or script, more lag between client and server may help increase the success rate)

after double-click, the loot content will unexpectedly closed，the ores dont get consumed, but it can not be Prospected again , until you get back to the "select character" menu and reenter the wow. You will find the ores will remain there, and you can Prospect and loot again.

**********************
UPDATE: This can be easily reproduced at high ping environment (about 1000ms) , just double click.

  

### Steps to Reproduce

1. learn 28895 (learn jewel craft)
2. learn 31252 (learn Prospect)
3. setskill 755 300 
4. add 10620 20
5. make a script or mouse marco so you can click very very (about 10ms between double click) soonly.( this is opt but will highly increase the success rate)
6. Or you can limit your netspeed,  to increase your ping the more the better, so you can do this by common double click
7. don't open "auto loot all item" (don't SHIFT-LOOT)
8. Prospect on a stack of ores, soonly double on ONE of the item. dont loot the last item.
9. after double-click, now you can not Prospect again, need to get back to the "select character" menu and re-enter the wow. 
10. you can loot at least one item from Prospecting, and the ores won't disappear. so you can get infinit items from limited ores.

### Expected behavior

https://github.com/cmangos/issues/assets/143975831/d5ee256d-3adb-478b-b79b-7d16e0a92f74

Like this, you can loot a azeroth diamond, but the ores are still there, and can be used as normal after re-logging in

### Core SHA1 Commit Hash

Core Release build 2023-9- 28

### Operating System

Windows 11

### Client Version

2.4.3 (The Burning Crusade)

**Comment by Nightingale9002** [comment]
In different version of cmangos core this bug can be found.

**Comment by DrKittens** [comment]
Can it be reproduced with only a single stack in your inventory?
If so when you prospect that single stack again can you loot another Azeroth Diamond or does it only show the remaining item?

Edit: What are your hashes?
Can you do ```git rev-parse head``` in the directories you cloned down for the database and core?
If you downloaded from release do you recall what release version you grabbed?

**Comment by Nightingale9002** [comment]
> 您的库存中只有一个堆栈可以复制吗？如果是这样，当您再次探索那一叠钻石时，您可以掠夺另一颗艾泽拉斯钻石还是只显示剩余的物品？

Yes , and it can produce  infinit diamonds,  only need to get back to the character menu, and re-log in. 
So, more stacks of ore  means lesser re-log.   Every time loot  new item.

**Comment by Nightingale9002** [comment]
> 您的库存中只有一个堆栈可以复制吗？如果是这样，当您再次探索那一叠钻石时，您可以掠夺另一颗艾泽拉斯钻石还是只显示剩余的物品？
> 
> 编辑：你的哈希是什么？您可以在为数据库和核心克隆的目录中执行操作吗？如果您从发行版下载，您是否还记得您抓取了哪个发行版？`git rev-parse head`


mangos-tbc-master.zip
SHA1 A7A798F8BA89C73F254197B5D1AB6B9437E6FC1D

I just  downloaded  this core version recently at 8/30,   but I have met this bug at different versions.

**Comment by DrKittens** [comment]
With a single stack of mithril i am unable to reproduce multiple pops of an Azeroth Diamond. 

For me the bug would be that after closing the prospecting window and relogging I retain my mithril thus can craft with it so the "cost" of prospecting is not paid properly.

I cant reproduce the reported issue where i can get repeated azeroth diamond drops from the same stack of mithril but I'd consider the bug i can reproduce (loot a prospecting reward, retain the materials used for the action) a bug that needs fixing regardless.

**Comment by Nightingale9002** [comment]
> 用一叠秘银，我无法重现艾泽拉斯钻石的多次爆裂。
> 
> 对我来说，错误在于，在关闭勘探窗口并重新记录后，我保留了我的秘银，因此可以用它制作，因此勘探的“成本”没有得到适当的支付。
> 
> 我无法重现报告的问题，我可以从同一堆秘银中获得重复的艾泽拉斯钻石掉落，但我认为我可以重现的错误（掠夺勘探奖励，保留用于行动的材料）无论如何都需要修复的错误。

I wonder if you really repeatly the routine  strictly. The most critical thing maybe 5.  I have tried to reproduce this bug by hand, but only once did. However it can be undoubtedly reproduced with a  mouse Marco like Razer synapse. Something like Python script also works.  I guess the thing is about if you can upload more than one quest of looting before the server's  respones getting back.

When mining and opening chest,  lootcontent will also close if you double click on one item . But these will still loot former remained items.

You don't need to close the content window by clicking the x or press ESC. The bug is when you quickly double click on the item in the loot window , the window closed. So server cannot automatically delete the ores.

**Comment by Nightingale9002** [comment]
@DrKittens  UPDATE: This can be easily reproduced at high ping environment, just double click.
just need to limit the net speed if your server is non-local.

**Comment by Nightingale9002** [comment]
![image](https://github.com/cmangos/issues/assets/143975831/6307b785-957f-4986-a066-1aa9c8bac6bd)

I have make this change to SpellEffect.ccp as a  stopgap. But for ALL loot windows after double-click it will close, I make a new issue for this. https://github.com/cmangos/issues/issues/3575
